src/alda/lisp/model/pitch.clj

```diff
@@ -17,17 +17,38 @@
   [note]
   (* 440.0 (Math/pow 2.0 (/ (- note 69.0) 12.0))))
 
+(defn- check-key
+  "Modifies the accidentals on notes to fit the key signature.
+   If there are no accidentals and this letter is in the signature, 
+   return the letter's signature accidentals, otherwise return 
+   existing accidentals"
+  [signature letter accidentals]
+   (if (empty? accidentals)
+       (get signature letter nil) 
+       (identity accidentals)))
+
 (defn pitch
   "Returns a fn that will calculate the frequency in Hz, within the context
    of the octave that an instrument is in."
-  [letter & accidentals]
-  (fn [octave & {:keys [midi]}]
+  [parsed-pitch]
+  (let [letter (first parsed-pitch) accidentals (rest parsed-pitch)]
+  (fn [octave key-signature & {:keys [midi]}]
     (let [midi-note (reduce (fn [number accidental]
+                              (log/debug (format "number %s accidental %s" number accidental))
```

Comment:
I should add some guidance about `log/debug` statements to our `CONTRIBUTING.md`, but to keep a long story short, I think it would be good to remove this one before we merge the pull request.

I think the general rule should maybe be this: debug statements should generally be avoided, but they are OK if it's something that you would want to see when you turn "debug mode" on to troubleshoot a problem. Right now there are debug statements for each note event that is played -- logging out the pitch and duration, etc. -- and to log that certain components (e.g. the MIDI synth) have loaded. Things like that. If we're adding debug statements to help us understand what data is being passed around while we're implementing some new code, then the debug statements should probably be taken out.

(cc: @crisptrutski )